src/frontmatter/frontmatter.ts

```diff
@@ -9,24 +9,41 @@ import { IFrontmatter, PropertyPath } from './types';
 import { countNewLines, shiftLines } from './utils';
 
 export class Frontmatter<T extends object = any> implements IFrontmatter<T> {
-  private readonly node: Unist.Literal | null;
-  private readonly root: Unist.Parent;
-  private readonly properties: T | null;
+  public readonly document: Unist.Parent;
+  private readonly node: Unist.Literal;
+  private properties: Partial<T> | null;
 
-  constructor(data: Unist.Parent | string) {
-    const root = typeof data === 'string' ? parseWithPointers(data).data : data;
+  constructor(data: Unist.Parent | string, dirty = true) {
```

Comment:
I truly believe we should mutate the data by default., as this is what most tools/plugins that modify ASTs do.

Reply:
Not sure on wether this is correct or not, but our tree is immutable (all updates go through patch system), so having other libs like this one mutate it themselves doesn't work.

Reply:
Yeah, in our case it's indeed immutable and given the fact this package is primarily used by Graphite, I believe we could make it immutable by default as opposed to the solution I proposed in the PR and still keep that flag if someone is willing to mutate the AST?

Reply:
Perfect! Not mutating is also generally "safer".